src/libs/meta/meta.c

```diff
@@ -1211,7 +1211,7 @@ int elektraSortTopology (KeySet * ks, Key ** array)
 					retVal = -1;
 					break;
 				}
-				else if (i == j)
+				else if ((long)i == (long)j)
 				{
```

Comment:
Is there no better solution than a cast?

Please also document that this branch is empty and purpose and that the case statement above has a fallthrough.